woocommerce/payment-gateway/payment-tokens/class-sv-wc-payment-gateway-payment-token.php

```diff
@@ -672,6 +672,39 @@ public function save() {
 	}
 
 
+	/**
+	 * Deletes a token from the database.
+	 *
+	 * Also deletes a token from the legacy user meta.
+	 * @see \WC_Payment_Token::delete()
+	 * @see SV_WC_Payment_Gateway_Payment_Tokens_Handler::delete_legacy_token()
+	 *
+	 * @since 5.6.0-dev.1
+	 *
+	 * @param bool $force_delete argument mapped to {@see \WC_Data::delete()}
+	 * @return bool whether a token was deleted (note: it will not evaluate deletion of legacy data)
+	 */
+	public function delete( $force_delete = false ) {
+
+		$deleted = false;
+
+		if ( $token = $this->get_woocommerce_payment_token() ) {
+			$deleted = $token->delete( $force_delete );
+		}
+
+		// delete legacy token in user meta data
```

Comment:
this follows a pattern as in `save()` method, which for now utilizes a tokens handler method to perform operations with legacy tokens